sn_transfers/src/wallet/local_store.rs

```diff
@@ -355,9 +355,12 @@ impl LocalWallet {
             }
         }
 
-        self.wallet
-            .payment_transactions
-            .extend(all_transfers_per_address);
+        // Add new payments to the existing map if present
+        // Use entry API to avoid multiple lookups in the BTreeMap
+        for (xorname, payment_details) in all_transfers_per_address {
```

Comment:
what's the benefit here by doing so ?
the previous `extend` doesn't filter out duplicated ones ? or it's too computation expensive ?

Reply:
I think we want to edit the underlying data and append to it. Directly calling extend in the upper layer might replace existing entries instead of merging them together.

Reply:
> Directly calling extend in the upper layer might replace existing entries instead of merging them together.

This is what appeared to be happening. 

Indeed this may be relted to the other issue you saw w/r/t price calculations